DragonSlayer.h

```diff
@@ -1,8 +1,19 @@
 #pragma once
+#include "Character.h"
+#include "AttackItem.h"
 
 struct DragonSlayer : Character
 {
+    DragonSlayer(std::string name_, int hitPoints_, int armor_);
+
+    const std::string& getName() override;
+    std::string getStats() override;
     void attack(Character& other) override;
+    AttackItem* attackItem = new AttackItem();
```

Comment:
Your'e storing heap allocated memory in a raw pointer. 
And you aren't freeing it either when the class goes out of scope. 
```
        //note: items are single-use only, so you need to reset it after use.  
        //look in the Character class for how the other item types are reset after use.
```
Take a look at how the other items are stored. 
Remember, you're only storing one, so a vector is the wrong choice.
- [ ] revise